src/System.Private.CoreLib/src/System/Text/StringBuilder.CoreCLR.cs

```diff
@@ -136,5 +136,33 @@ internal unsafe void InternalCopy(IntPtr dest, int len)
             }
             while (currentSrc != null);
         }
+
+        /// <summary>
+        /// Checks if this StringBuilder contains only one chunk (char buffer).
+        /// </summary>
+        /// <param name="chunk">A reference to the internal chunk.</param>
+        /// <returns>Returns <c>true</c> if this StringBuilder only has one chunk; otherwise false.</returns>
+        /// <remarks>
+        /// This method is used by the marshalling system to determine if this StringBuilder can be passed to native
+        /// by only having its buffer pinned.
+        /// </remarks>
+        internal bool IsSingleChunk(out char[] chunk)
```

Comment:
`TryGetSingleChunk` is a name more consistently used for pattern. I wouldn't expect a method named `IsXXX` to have an out parameter.

Reply:
But you don't need the chunk if it's not the only chunk, right? Why not just return the chunk as the return parameter, and use null if it's not the only one? The IL might be a little different, but eliminating the out parameter might generate more compact code

Reply:
Normally I am loathe to use `null` as an indication, but in this case the scenario is for an internal marshaller so I think this is a good idea. @jkoritzinsky what are you thoughts?

Reply:
I think since it's for internal use only that it's worth trying out. I'm currently working on it. Makes some of the generated IL cleaner, so I think it's worth doing.